pulsar/producer_partition.go

```diff
@@ -1446,6 +1431,22 @@ type sendRequest struct {
 	reservedMem      int64
 }
 
+func (sr *sendRequest) done(msgID MessageID, err error) {
+	sr.callbackOnce.Do(func() {
+		runCallback(sr.callback, msgID, sr.msg, err)
+	})
+
+	if sr.transaction != nil {
+		sr.transaction.endSendOrAckOp(err)
+	}
+
+	if sr.totalChunks <= 1 || sr.chunkID == sr.totalChunks-1 {
```

Comment:
In @gunli's PR, all these three lines are in this if block. I defer this change until we really need it.

@gunli may you provide some background why we need to call `runCallback` and `endSendOrAckOp` with this condition?